src/account/twoFactor/useTwoFaToggleMutation.js

```diff
@@ -0,0 +1,81 @@
+import { useDataMutation, useAlert } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { useMemo, useState } from 'react'
+import userProfileStore from '../../profile/profile.store.js'
+
+const mutationBase = {
+    type: 'create',
+    data: ({ code }) => ({ code }),
+}
+const enableMutation = {
+    ...mutationBase,
+    resource: '/2fa/enabled',
+}
+const disableMutation = {
+    ...mutationBase,
+    resource: '/2fa/disabled',
+}
+
+const getAlertMessage = ({ nextIsTwoFaEnabled, error }) => {
+    if (nextIsTwoFaEnabled) {
+        if (error) {
+            return (
+                error?.message ??
+                i18n.t('Could not enable 2 Factor Authentication')
+            )
+        } else {
+            return i18n.t('Two factor authentication was enabled successfully')
+        }
+    } else {
+        if (error) {
+            return (
+                error?.message ??
+                i18n.t('Could not disable two factor authentication')
+            )
+        } else {
+            return i18n.t('Two factor authentication was disabled successfully')
+        }
+    }
+}
+
+const getAlertOptions = ({ error }) =>
+    error ? { critical: true } : { success: true }
+
+export default function useTwoFaToggleMutation({ isTwoFaEnabled }) {
+    const [
+        lastActionWasTwoFaDisableSuccess,
+        setLastActionWasTwoFaDisableSuccess,
+    ] = useState(false)
+    const { show: showAlert } = useAlert(getAlertMessage, getAlertOptions)
+    const mutationOptions = useMemo(() => {
+        const nextIsTwoFaEnabled = !isTwoFaEnabled
+        return {
+            onComplete: () => {
+                userProfileStore.state.twoFaEnabled = nextIsTwoFaEnabled
+                userProfileStore.setState(userProfileStore.state)
```

Comment:
I'm not exactly sure, it could be possible to do the following:
```js
userProfileStore.setState({
    ...userProfileStore.state,
    twoFaEnabled: attemptingToEnableTwoFa,
})
```
But this seems to be the idiomatic way to do it with the old `Store` in `d2-ui`: first mutate then call `setState` with the mutated object as a parameter....